File: kotlin/pangram/src/main/kotlin/Pangram.kt
object Pangram {

    fun isPangram(input: String): Boolean {
        val allCharacters = ('a'..'z').toMutableSet()
        input.forEach { allCharacters.remove(it.toLowerCase()) }
        return allCharacters.isEmpty()

        // alt solution
        // val allCharacters = ('a'..'z').toSet()
        // val inputCharacters = input.toLowerCase().toSet()
        // return (allCharacters - inputCharacters).isEmpty()
    }
}

File: go/pangram/pangram.go
package pangram

import (
	"unicode"
)

func IsPangram(input string) bool {
	var histogram = map[rune]int{}
	for r := 'a'; r <= 'z'; r++ {
		histogram[r] = 0
	}

	for _, r := range input {
		l := unicode.ToLower(r)
		if !unicode.IsLetter(l) {
			continue
		}
		histogram[l]++
	}

	for _, v := range histogram {
		if v == 0 {
			return false
		}
	}

	return true
}
